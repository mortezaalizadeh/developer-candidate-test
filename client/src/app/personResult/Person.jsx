import React from 'react';
import PropTypes from 'prop-types';
import TableCell from '@material-ui/core/TableCell';
import TableRow from '@material-ui/core/TableRow';
import Checkbox from '@material-ui/core/Checkbox';
import { PersonPropType } from './PropTypes';

const Person = ({ onClick, isSelected, person: { _id, name, age, gender } }) => (
  <TableRow hover onClick={event => onClick(event, _id)} role="checkbox" aria-checked={isSelected} tabIndex={-1} key={_id} selected={isSelected}>
    <TableCell padding="checkbox">
      <Checkbox checked={isSelected} />
    </TableCell>
    <TableCell component="th" scope="row" padding="none">
      {name}
    </TableCell>
    <TableCell align="right">{gender}</TableCell>
    <TableCell align="right">{age}</TableCell>
  </TableRow>
);

Person.propTypes = {
  onClick: PropTypes.func.isRequired,
  isSelected: PropTypes.bool.isRequired,
  person: PersonPropType.isRequired,
};

export default Person;
