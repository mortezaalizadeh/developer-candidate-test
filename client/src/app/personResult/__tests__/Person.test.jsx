import React from 'react';
import renderer from 'react-test-renderer';
import Person from '../Person';

it('should render correctly', () => {
  const person = { _id: 'person  unqiue id', name: 'Morteza Alizadeh', age: 37, gender: 'male' };
  const tree = renderer.create(<Person person={person} />).toJSON();

  expect(tree).toMatchSnapshot();
});
