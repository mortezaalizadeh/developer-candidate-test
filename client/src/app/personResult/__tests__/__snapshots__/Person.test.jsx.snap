// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<tr
  className="MuiTableRow-root-1 MuiTableRow-hover-3"
  onClick={[Function]}
  role="checkbox"
  tabIndex={-1}
>
  <td
    aria-sort={null}
    className="MuiTableCell-root-6 MuiTableCell-paddingCheckbox-12"
  >
    <span
      className="MuiButtonBase-root-34 MuiIconButton-root-28 MuiPrivateSwitchBase-root-24 MuiCheckbox-root-18 MuiCheckbox-colorSecondary-23"
      onBlur={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={null}
    >
      <span
        className="MuiIconButton-label-33"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-37"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
          />
        </svg>
        <input
          className="MuiPrivateSwitchBase-input-27"
          data-indeterminate={false}
          onChange={[Function]}
          type="checkbox"
        />
      </span>
      <span
        className="MuiTouchRipple-root-46"
      />
    </span>
  </td>
  <th
    aria-sort={null}
    className="MuiTableCell-root-6 MuiTableCell-paddingNone-13"
    scope="row"
  >
    Morteza Alizadeh
  </th>
  <td
    aria-sort={null}
    className="MuiTableCell-root-6 MuiTableCell-alignRight-16"
  >
    male
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root-6 MuiTableCell-alignRight-16"
  >
    37
  </td>
</tr>
`;
